node/packages/aws-lambda-sdk/instrument/index.js

```diff
@@ -28,6 +28,31 @@ const { awsLambda: awsLambdaSpan, awsLambdaInitialization: awsLambdaInitializati
 
 const toProtobufEpochTimestamp = awsLambdaSpan.constructor._toProtobufEpochTimestamp;
 
+const resolveBodyString = (data, prefix) => {
+  if (data === undefined) return null;
+  data = { ...data };
+  if (awsLambdaSpan.tags.get('aws.lambda.event_source') === 'aws.apigateway') {
+    if (typeof data.body === 'string') {
+      if (data.isBase64Encoded) {
+        delete data.body;
+        serverlessSdk._reportNotice('Binary body excluded', `${prefix}_BODY_BINARY`, {
+          _traceSpan: awsLambdaSpan,
+        });
+      }
+    }
+  }
+  if (
+    data.body &&
+    Buffer.byteLength(JSON.stringify(data.body)) > serverlessSdk._maximumBodyByteLength
+  ) {
```

Comment:
Wouldn't we also want to check if `data` by itself is too large? I am thinking of the scenario when you might be passing in a custom object as an input to your lambda function.

Reply:
@Danwakeem that's a very good point. We should actually validate the whole request size, and if it's too large, return null. I've updated PR

Reply:
Ahh okay awesome just left one more comment 🙏